controllers/snapshotenvironmentbinding_controller.go

```diff
@@ -183,6 +183,23 @@ func detectTargetFromEnvironment(ctx context.Context, environment appstudiov1alp
 	}
 }
 
+func remoteSecretApplicableForEnvironment(remoteSecret rapi.RemoteSecret, environmentName string) bool {
+	var targetEnvironments []string
+	if environmentLabelInSecret, labelSet := remoteSecret.Labels[EnvironmentLabelName]; labelSet {
+		targetEnvironments = []string{environmentLabelInSecret}
```

Comment:
```suggestion
		return targetEnvironments == environmentLabelInSecret
```
Imho this could be cleaner